pkg/adaptor/cloud/libvirt/libvirt_test.go

```diff
@@ -0,0 +1,111 @@
+// (C) Copyright Confidential Containers Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package libvirt
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/confidential-containers/cloud-api-adaptor/pkg/adaptor/cloud"
+	"github.com/stretchr/testify/assert"
+	libvirtxml "libvirt.org/go/libvirtxml"
+)
+
+var testCfg Config
+
+func init() {
+	cloud.DefaultToEnv(&testCfg.URI, "LIBVIRT_URI", "") // explicitly no fallback here
+	cloud.DefaultToEnv(&testCfg.PoolName, "LIBVIRT_POOL", defaultPoolName)
+	cloud.DefaultToEnv(&testCfg.NetworkName, "LIBVIRT_NET", defaultNetworkName)
+	cloud.DefaultToEnv(&testCfg.VolName, "LIBVIRT_VOL_NAME", defaultVolName)
+}
+
+func checkConfig(t *testing.T) {
+	if testCfg.URI == "" {
+		t.Skipf("Skipping because LIBVIRT_URI is not configured")
+	}
+}
+
+func TestLibvirtConnection(t *testing.T) {
+	checkConfig(t)
+
+	client, err := NewLibvirtClient(testCfg)
+	if err != nil {
+		t.Error(err)
+	}
+	defer client.connection.Close()
+
+	assert.NotNil(t, client.nodeInfo)
+	assert.NotNil(t, client.caps)
+}
+
+func TestGetArchitecture(t *testing.T) {
+	checkConfig(t)
+
+	client, err := NewLibvirtClient(testCfg)
+	if err != nil {
+		t.Error(err)
+	}
+	defer client.connection.Close()
+
+	node, err := client.connection.GetNodeInfo()
+	if err != nil {
+		t.Error(err)
+	}
+
+	arch := node.Model
+	if arch == "" {
+		t.FailNow()
+	}
+}
+
+func verifyDomainXML(domXML *libvirtxml.Domain) error {
+	arch := domXML.OS.Type.Arch
+	if arch != "s390x" {
+		return nil
```

Comment:
Could it skip instead of pass if not in s390x? Just add a skip on  TestCreateDomainXMLs390x() if not s390x...